src/swarm/neo/client/mixins/SuspendableRequestCore.d

```diff
@@ -0,0 +1,1165 @@
+    private RequestEventDispatcher* request_event_dispatcher;
+
+    /// Fiber to suspend when waiting for I/O.
+    private MessageFiber fiber;
+
+    /***************************************************************************
+
+        Performs the logic of awaiting state change signals from the user-facing
+        controller class (see SuspendableController) and sending the
+        corresponding messages to the node.
+
+        Returns only when the request has been stopped by the user.
+
+        Params:
+            conn = request-on-conn event dispatcher
+            context = request context
+            request_event_dispatcher = multi-fiber request event dispatcher
+            fiber = fiber to suspend when waiting for I/O
+
+    ***************************************************************************/
+
+    public void handle ( RequestOnConn.EventDispatcherAllNodes conn,
+        Request.Context* context, RequestEventDispatcher* request_event_dispatcher,
+        MessageFiber fiber )
+    {
+        this.conn = conn;
+        this.context = context;
+        this.request_event_dispatcher = request_event_dispatcher;
+        this.fiber = fiber;
+
+        auto suspendable_control =
+            &this.context.shared_working.suspendable_control;
+
+        bool stopped, state_changed_in_notifier;
+        do
+        {
+            if ( !state_changed_in_notifier )
+            {
+                // Wait for user action.
+                auto event = this.request_event_dispatcher.nextEvent(this.fiber,
+                    Signal(suspendable_control.Signal.StateChangeRequested));
+                assert(event.active == event.active.signal);
+                assert(event.signal.code == suspendable_control.Signal.StateChangeRequested);
+            }
+
+            // Send state change message to node and wait for ACK.
+            stopped = this.changeRequestState();
+
+            // Notify the user that the state change has taken effect and handle
+            // the case where they request a new state change in the notifier.
+            state_changed_in_notifier = this.notifyUser(stopped);
+        }
+        while ( !stopped );
+    }
+
+    /***************************************************************************
+
+        After a state change has been carried out by the node, notifies the user
+        if this was the last request-on-conn to carry out the state change. Also
+        checks for the case that the user initiates another state change in the
+        notifier call (see return value).
+
+        Params:
+            stopped = indicates whether the just ACKed state change was to stop
+                the request (true) or was another type of state change (false)
+
+        Returns:
+            true if another state change was requested during notification
+
+    ***************************************************************************/
+
+    private bool notifyUser ( bool stopped )
```

Comment:
I was trying to see how this flag is used, but it seems that it's not used at all (less complexity, which is good).